pytest show ImportError

### Version Checks (indicate both or one)

- [x] I have confirmed this bug exists on the lastest [release](https://github.com/pypsa/pypsa/releases) of PyPSA.

- [x] I have confirmed this bug exists on the current [`master`](https://github.com/pypsa/pypsa/tree/master) branch of PyPSA.


### Issue Description

I created a fork from pypsa and cloned it to my computer. 
Then I followed the contribution doc and installed the dependencies with

pip install -e .[dev]

After that, running pytest results in an ImportError.

=> Maybe some dependencies are not correctly specified. 

```
C:\python_env\workspace\pypsa git:[master]
pytest
ImportError while loading conftest 'C:\python_env\workspace\pypsa\test\conftest.py'.
test\conftest.py:10: in <module>
    import geopandas as gpd
..\..\App\WinPython\python-3.12.4.amd64\Lib\site-packages\geopandas\__init__.py:3: in <module>
    from geopandas.geoseries import GeoSeries
..\..\App\WinPython\python-3.12.4.amd64\Lib\site-packages\geopandas\geoseries.py:9: in <module>
    import pandas as pd
..\..\App\WinPython\python-3.12.4.amd64\Lib\site-packages\pandas\__init__.py:27: in <module>
    __import__(_dependency)
..\..\App\WinPython\python-3.12.4.amd64\Lib\site-packages\pytz\__init__.py:20: in <module>
    from pytz.tzinfo import unpickler, BaseTzInfo
..\..\App\WinPython\python-3.12.4.amd64\Lib\site-packages\pytz\tzinfo.py:27: in <module>
    _epoch = datetime.utcfromtimestamp(0)
E   DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
```

### Reproducible Example

```python
pytest
```

### Expected Behavior

Run tests for a fresh clone without issues

### Installed Versions

<details>
Replace this line.
</details>

--- a/test/test_docs.py
+++ b/test/test_docs.py
@@ -13,6 +13,13 @@ import pytest
 
 import pypsa
 
+try:
+    import cartopy  # noqa
+
+    cartopy_available = True
+except ImportError:
+    cartopy_available = False
+
 # Warning: Keep in sync with settings in doc/conf.py
 doctest_globals = {
     "numpy": np,
@@ -32,6 +39,7 @@ modules = [
     sys.version_info[:2] == (3, 10),
     reason="Doctest fail until linopy supports numpy 2 on all python versions",
 )
+@pytest.mark.skipif(not cartopy_available, reason="Cartopy not available")
 @pytest.mark.parametrize("module", modules)
 def test_doctest(module):
     finder = doctest.DocTestFinder()

